subsys/bluetooth/mesh/light_temp_srv.c

```diff
@@ -394,14 +407,29 @@ static void light_temp_srv_reset(struct bt_mesh_light_temp_srv *srv)
 	srv->range.max = BT_MESH_LIGHT_TEMP_MAX;
 }
 
+static int update_handler(const struct bt_mesh_model *model)
+{
+	struct bt_mesh_light_temp_srv *srv = model->rt->user_data;
+	struct bt_mesh_light_temp_status status = { 0 };
+
+	temp_get(srv, NULL, &status);
```

Comment:
```suggestion
	srv->handlers->get(srv, NULL, &status);
```
Seems `temp_get` does nothing, just copies data to some local variable to move it again to another local variable.

Reply:
I have tried to have it similar to `subsys/bluetooth/mesh/light_ctl_srv.c` where `ctl_get` is called which gets a superset of the values of `temp_get`. I have tested the code in production and it allowed to sent publication period and published correct values in set interval. If it really does nothing maybe the ligth_ctl code has to be adapted?

Can not check on my production system right now will be at the office back on tuesday.

```C
static void ctl_get(struct bt_mesh_light_ctl_srv *srv,
		    struct bt_mesh_msg_ctx *ctx,
		    struct bt_mesh_light_ctl_status *status)
{
	struct bt_mesh_light_temp_status temp;
	struct bt_mesh_lightness_status light;

	srv->lightness_srv.handlers->light_get(&srv->lightness_srv, ctx,
					       &light);
	srv->temp_srv.handlers->get(&srv->temp_srv, ctx, &temp);

	status->current_temp = temp.current.temp;
	status->current_light = light.current;
	status->target_temp = temp.target.temp;
	status->target_light = light.target;
	status->remaining_time = MAX(temp.remaining_time, light.remaining_time);
}

static int update_handler(const struct bt_mesh_model *model)
{
	struct bt_mesh_light_ctl_srv *srv = model->rt->user_data;
	struct bt_mesh_light_ctl_status status = { 0 };

	ctl_get(srv, NULL, &status);
	ctl_encode_status(srv->pub.msg, &status);

	return 0;
}
```

Reply:
I do not think that the referencing to CTL model implementation is correct here. 
Mesh model spec 1.1
```
6.4.3 Light CTL Server
6.4.3.1 Description
...
The Light CTL Server is a main model that extends the Light Lightness Server model (see Section
6.4.1) and corresponds with the Light CTL Temperature Server model (see Section 6.4.4).
...
```
BLE Mesh spec 1.1
```
2.3.6 Models
...
Correspond. When models share state/procedure instances and work together to achieve some
functionality, they have a correspond relationship. Models that have a correspond relationship may
share either state instances or state bindings; however, they may have different messages and
uninherited procedures interacting with the states.
...
```
CTL and CTL Temperature servers share states\procedures. To generate its own status, CTL server has to use CTL Temperature server to pick up temperature values.

To create its own status, CTL Temperature server doesn't require any other servers. It can call its own `get` handler directly.

That means the presence of `ctl_get` function is reasonable, but `temp_get` really does nothing here. Just call `get` handler in `update_handler`.
I'm quite sure it will work the same but without unnecessary wrapper function. :)